src/runtime/compact.cpp

```diff
@@ -379,7 +379,7 @@ compacted_region::~compacted_region() {
         m_nested_mpzs->m_value.~mpz();
         m_nested_mpzs = *reinterpret_cast<mpz_object**>(reinterpret_cast<char*>(m_nested_mpzs) + sizeof(mpz_object));
     }
-    free(m_begin);
+    delete[] reinterpret_cast<char*>(m_begin);
```

Comment:
This looks wrong. We use `malloc` on line 370. Are we using `new` somewhere?

Reply:
Good catch, I changed this allocation instead: https://github.com/leanprover/lean4/pull/157/files#diff-ae4f52a26f45f3835b747fe8b1d53b71R82-R83

Reply:
> Good catch, I changed this allocation instead: https://github.com/leanprover/lean4/pull/157/files#diff-ae4f52a26f45f3835b747fe8b1d53b71R82-R83

Yes, this is the correct fix.